### ISSUE TITLE
v10.9.0 regression: Vue files fail with 'Parsing error: Unexpected token <' due to broken vue-eslint-parser reference in flat config

### ISSUE THREAD

**Issue by zoispag** [vuejs/eslint-plugin-vue#3076]
### Description

After upgrading from `eslint-plugin-vue` v10.8.0 to v10.9.0, ESLint fails to parse **all** `.vue` files with:

```
  1:1  error  Parsing error: Unexpected token <
```

This is a regression introduced by the TypeScript migration of configs in v10.9.0 ([#3002](https://github.com/vuejs/eslint-plugin-vue/pull/3002)).

### Root Cause

In `dist/configs/flat/base.js` (v10.9.0), the compiled output does:

```js
let vue_eslint_parser = require("vue-eslint-parser");
vue_eslint_parser = require_runtime.__toESM(vue_eslint_parser);

// ...
languageOptions: {
  parser: vue_eslint_parser.default,  // <-- undefined!
  sourceType: "module"
},
```

The problem is that `vue-eslint-parser` is a CommonJS module that sets `__esModule: true` on its exports (a TypeScript compilation artifact) but does **not** export a `default` property:

```js
const vueParser = require('vue-eslint-parser');
// vueParser.__esModule === true
// vueParser.default === undefined  (!)
// vueParser.parse === function  (the actual parser)
```

The rolldown `__toESM` helper sees `__esModule: true` and skips the `{ default: mod }` wrapping, so `vue_eslint_parser.default` is `undefined`. The Vue parser is never applied to `.vue` files.

**In v10.8.0**, the config used a direct `require`:
```js
// dist/configs/flat/base.js (v10.8.0)
parser: require("vue-eslint-parser"),  // worked fine
```

### Reproduction

```js
const p = require('eslint-plugin-vue');
const cfg = p.configs['flat/essential'];
const parserConfig = cfg[1]; // the config with files: ['*.vue', '**/*.vue']
console.log(parserConfig.languageOptions.parser); // undefined in 10.9.0, parser object in 10.8.0
```

### Versions

- `eslint-plugin-vue`: 10.9.0 (broken), 10.8.0 (works)
- `vue-eslint-parser`: 10.2.0
- `eslint`: 9.39.4
- Node.js: 24.15.0

### Fix Suggestion

In the TypeScript source (`lib/configs/flat/base.ts`), change:

```ts
import vueParser from 'vue-eslint-parser'
```

to:

```ts
import * as vueParser from 'vue-eslint-parser'
```

Or handle the CJS/ESM interop explicitly so the compiled output uses the module directly rather than `.default`.

**Comment by FloEdelmann** [comment]
`vue-eslint-parser` is a peer dependency of `eslint-plugin-vue`, so you should be able to update it manually yourself to v10.3.0. Could you please check whether that fixes it? If so, then #3075 would be the correct fix for this repo (increase our minimum supported version of `vue-eslint-parser`).

**Comment by zoispag** [comment]
Confirmed. Upgrading `vue-eslint-parser` to `^10.3.0` fixes the issue. CI is green with `eslint-plugin-vue@10.9.0` + `vue-eslint-parser@10.4.0` (latest `^10.3.0`).

So yes, #3075 is the right fix for this repo.

**Comment by FloEdelmann** [comment]
Thank you for testing! I released https://github.com/vuejs/eslint-plugin-vue/releases/tag/v10.9.1 with the fix.
